mantidproject/mantid
# DecouplingAsymPowderMag Fitting function for Muons

**Description of work.**

Added two new fitting functions for Decoupling of asymmetry in the ordered state of a powdered magnet fitting:-

- DecoupAsymPowderMagLong for longitudinal polarization
- DecoupAsymPowderMagRot for rotational asymmetry

Both functions use the same underlying formula and so a helper file has been added to provide this functionality to both fitting functions.

Includes tests for the helper and fitting functions and revelvant documentation.

**To test:**
1. Download [red_data.csv](https://github.com/mantidproject/mantid/files/7676093/red_data.csv)
2. Open Mantid
3. Load the red_data.csv file
4. Double click the workspace that loads and a plot should appear
5. Click Fit
6. Add a FlatBackground fitting function 
7. Add a DecoupAsymPowderMagLong fitting function and set the CharField at 1196 and fix it.
8. Run the fit

Fixes #32585 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
